### ISSUE TITLE
GPX not applying profiles under Cura

### ISSUE THREAD

**Issue by stuartpb** [stuartpb/repligreater#2]
From an email I just got (with light adjustments for formatting):

> Stuart:
> 
>   Thank you for taking on the pain of figuring out how to get a QIDI Tech I
> profile for Cura. I've been trying to get My QTI working for a little bit
> especially printing a 100% infill calibration cube. The printer kept trying
> to extrude too much material in the infil (walls were fine). I finally
> looked at the Cura logfiles and noticed that it wasn't applying a printer
> profile in GPX when X3GWriter fires off the conversion.
> 
> I don't have a Github account and didn't want to sign up just for a simple
> request, so I'll put it in e-mail here.  For some reason, in Cura 3.6.0, it
> seems that GPX was ignoring the `"@printer r1d\n;"` macro.
> 
> If I put:
>
> ```json
> "file_formats": "application/x3g",
> ```
>
> in the metadata section of the qiditech_i.def.json file, I get the correct
> detection in the log file of this:
>
> ```
> 2018-12-16 22:47:54,466 - DEBUG - [Thread-1] X3GWriter.X3GWriter.write
> [50]: Using configured machine: r1d (Replicator 1 - dual extruder)
> ```
> 
> Whereas before it wouldn't print anything about which printer it was
> using.  The prints appear to be working better now.
> 
> Thoughts?  Open to discussion.
> 
>   Regards,
>   Jeff

Here's the reply I gave:

> Sounds like a weird bug in Cura / the GPX plugin (might be a regression, I think the last Cura I used these on was something like 3.2.0). I'll copy this email to the issues on the repligreater repo, then file a cross-reference issue over in Cura's repo (and I'll send a followup with those issue numbers).
>
> Just as a heads up, though, you're probably going to need to create a GitHub account for further comments on this bug, since I won't really be able to investigate further, as I don't really use my QTI any more (the dual nozzles were ruining more prints with head crashes than they were helping with multi-material prints). I've replaced it with a Prusa i3 MK3, which has much broader support, only cost about a hundred bucks more (as a kit), and even has its own manufacturer-developed open-source slicer (Slic3r Prusa Edition) that doesn't give me any of these headaches I kept having in trying to use MakerBot-derived designs like this one.

**Comment by Ghostkeeper** [comment]
I'm not able to reproduce any variant behaviour dependent on `file_formats` in the metadata as this user describes. Note that the QTI profile already contains the `file_formats` metadata by default that Jeff said he added.

Removing the `file_formats` entry from the metadata caused g-code to be selected as the default file format in Cura but when I selected X3G there it created exactly the same file except a timestamp at the beginning (bytes 15 through 22; the same bytes change if I save it a third time without changing the metadata).

I'm not seeing the log entry either though. That is a log entry that's written by X3GWriter when it [finds the `machine_x3g_variant` metadata entry in the machine definition](https://github.com/Ghostkeeper/X3GWriter/blob/d76dc5e965ff2075a5b8a3706e1c5c362d417b57/X3GWriter.py#L160). The definition in this repository doesn't have it, so it makes sense that I'm not seeing the log entry, with or without the `file_formats` metadata entry.

X3GWriter doesn't look at the g-code (or the start g-code) for that `@printer` macro. That's something that GPX apparently does. Perhaps Jeff was using different versions of X3GWriter, where one of them at some point specified the R1D GPX profile, and this caused GPX to ignore the `@printer` macro? X3GWriter used to select R1D by default because that's what was needed for the Malyan printer that I originally rewrote the plug-in for.

In short, I suspect that other things were changed as well than just the `file_formats` metadata entry.
